app/assess/templates/macros/flag_history.html

```diff
@@ -0,0 +1,86 @@
+    {% set items = [] %}
+    {% for data in flag_data.updates %}
+        {% set FlagDetails %}
+            {% if data['status']=='RAISED' %}
+                <p><strong>Reason</strong><br>{{ data['justification'] }}</p>
+                <p><strong>Section(s) flagged</strong><br></p>
+                {{ display_flagged_sections(state, application_id, flag_data['sections_to_flag']) }}
+                <p><strong>Flag allocation</strong><br> <span class="flagged-tag">Flagged for {{ data['allocation'] }}</span></p>
+                <p><strong>Notification sent</strong><br>No</p>
+            {% else %}
+                {% if data['status']=='STOPPED' %}
+                    <p><strong>Resolve flag action</strong><br>Stop assessment</p>
+                {% elif data['status']=='ESCALATED' %}
+                    <p><strong>Resolve flag action</strong><br>Flag escalated to {{ data['allocation'] }}</p>
+                {% elif data['status']=='RESOLVED' %}
+                    <p><strong>Resolve flag action</strong><br>Query resolved</p>
+                {% endif %}
+                <p><strong>Reason</strong><br>{{ data['justification'] }}</p>
+            {% endif %}
+
+        {% endset %}
+
+        {% set username_date = '('+ accounts_list[data['user_id']]['full_name'] + ')<br>'+ data['date_created'] | utc_to_bst %}
+        {% set CreatedBy = username_date if data['status']=='RAISED' else 'N/A' %}
+        {% set ResolvedBy = username_date if data['status']!='RAISED' else 'N/A' %}
+
+        {% set addItem = items.append([{'html': CreatedBy},
+                                        {'text': FlagDetails},
+                                        {'html': ResolvedBy},
+                                        {'text': data['status'] | capitalize}]) %}
+    {% endfor %}
+
+    <h2 class="govuk-heading-l">Flagged for {{ flag_data.allocation }}</h2>
+    {{ govukTable({
+    'head': [
+        {'text': "Created by"},
+        {'text': "Flag details"},
+        {'text': "Resolved by"},
+        {'text': "Flag status"}
+    ],
+    'rows': items
+    })}}
+{% endmacro %}
+
+{% macro flag_tabs(state, application_id, flags_list, accounts_list) %}
+    {% set TabsData %}
+        {% set items = [] %}
+        {% for flag_item in flags_list %}
+            {% set addItem = items.append({
+            'label': "Flagged for " + flag_item.allocation,
+            'id': "tab-" + flag_item.allocation,
+            'panel': {
+                'html': flag_table(state, application_id, flag_item, accounts_list)}
+            })%}
+        {% endfor %}
+
+        {{ govukTabs({
+        'items': items
+        }) }}
+    {% endset %}
+
+    <details class="govuk-details" data-module="govuk-details" open="">
+        <summary class="govuk-details__summary">
+        <span class="govuk-details__summary-text">
+            Flag history
+        </span>
+        </summary><br><br>
+        {{ TabsData }}
+    </details>
+
+{% endmacro %}
```

Comment:
Ignore me if all these macros belong to the flag history otherwise, do you think we could have them in separate files?

Reply:
all macros belongs to flags history. Only `display_flagged_sections` is reusable